packages/lesswrong/server/callbacks/userCallbacks.ts

```diff
@@ -195,3 +198,65 @@ getCollectionHooks("Users").editSync.add(async function usersEditCheckEmail (mod
   }
   return modifier;
 });
+
+getCollectionHooks("Users").editAsync.add(async function subscribeToForumDigest (newUser: DbUser, oldUser: DbUser) {
+  if (
+    isAnyTest ||
+    forumTypeSetting.get() !== 'EAForum' ||
+    newUser.subscribedToDigest === oldUser.subscribedToDigest
+  ) {
+    return;
+  }
+
+  const mailchimpAPIKey = mailchimpAPIKeySetting.get();
+  const mailchimpForumDigestListId = mailchimpForumDigestListIdSetting.get();
+  const { lat: latitude, lng: longitude } = userGetLocation(newUser);
+  const status = newUser.subscribedToDigest ? 'subscribed' : 'unsubscribed';   
+  await fetch(`https://us8.api.mailchimp.com/3.0/lists/${mailchimpForumDigestListId}/members`, {
+    method: 'POST',
+    body: JSON.stringify({
+      email_address: newUser.email,
+      email_type: 'html', 
+      location: {
+        latitude,
+        longitude,
+      },
+      merge_fields: {
+        FNAME: newUser.fullName
+      },
+      status,
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `API_KEY ${mailchimpAPIKey}`,
+    },
+  });
+});
+
+getCollectionHooks("Users").newAsync.add(async function subscribeToDripCampaign(user: DbUser) {
+  if (isAnyTest || forumTypeSetting.get() !== 'EAForum') {
+    return;
+  }
+  const mailchimpAPIKey = mailchimpAPIKeySetting.get();
+  const mailchimpEAForumListId = mailchimpEAForumListIdSetting.get();
+  const { lat: latitude, lng: longitude } = userGetLocation(user);
+  await fetch(`https://us8.api.mailchimp.com/3.0/lists/${mailchimpEAForumListId}/members`, {
+    method: 'POST',
+    body: JSON.stringify({
+      email_address: user.email,
+      email_type: 'html', 
+      location: {
+        latitude,
+        longitude,
+      },
+      merge_fields: {
+        FNAME: user.fullName
+      },
+      status: "subscribed",
+    }),
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `API_KEY ${mailchimpAPIKey}`,
+    },
+  });
+});
```

Comment:
I'm awaiting confirmation from Aaron that I've understood the intended functionality of this drip campaign section correctly, so possibly it will be changed or removed once I hear back. Unless it gets removed, I expect the changes to primarily happen within Mailchimp.

Reply:
Followup: this is not immediately going to be connected to a drip campaign, but Aaron was in favor of adding new users to this list anyway for a forthcoming set of emails.